### PR TITLE
Multisample input using design file

### PR DESCRIPTION
**Pull request by Vlad-Dembrovskyi**

### CODE REVIEW

**Review by R-Cardenas — DISMISSED** [review]

**Comment by R-Cardenas** [comment]
## Overview

This PR allows the exomiser pipeline to accept multiple samples in the form of a design file. The design file contains the location of the VCF files and family information. The family information is used by the first process to create 3 files:

- ID file (contains proband ID for VCF)
- PED file (contains family information for samples within the multi sample VCF)
- HPO file (the phenotype IDs the user is interested in)


## Purpose

To achieve that two things were done:

1) Accept a CSV file that is then split by row.
2) Create a process that will create the files we need.

## Acceptance Criteria/Jira Ticket

[DEL-11750]()

## Changes

- Implements families_file parameter (for the design file)
- Removes single vcf input
- Add parameter for python module to create ped Files
- Add process that accepts design file
- Create channel that contains tuple values for PED,HPO,ID and VCFs.

## Internal CloudOS Tests
> Not necessary if run in Internal CloudOS CI
- JOB ID: 6398d4005f2b5e015b70c523 Workspace: RR603



## PR Checklist

#### Author to check:

- [Y] The PR title is informative and begins with appropriate `[Fix:|Feat:|Docs:]` based on PR type
- [Y] The description contains the acceptance criteria or link to the associated Jira ticket
- [Y] The description contains links to CloudOS tests covering all test cases in client environment

#### Reviewer to check:

- [ ] A new test profile covering new functions for acceptance criteria is created if it does not already exist
- [ ] Nextflow configs have associated profiles named to match the config basename
- [ ] The [Jenkinsfile](https://github.com/lifebit-ai/<replace-with-repo-name>/blob/dev/Jenkinsfile) has been updated with current Nextflow profiles
- [ ] The [internal_lifebit_cloudos_ci.yml](https://github.com/lifebit-ai/<replace-with-repo-name>/blob/dev/.github/workflows/internal_lifebit_cloudos_ci.yml) has been updated to run all current Nextflow profiles
- [ ] Redundant tests have been removed
- [ ] The [README.md](https://github.com/lifebit-ai/<replace-with-repo-name>/blob/dev/docs/README.md) has been updated with parameter changes and current `results/` and conforms to the README.md guidelines
- [ ] The code is logical, readable, reasonably optimal and meets the acceptance criteria
- [ ] The [containers](https://github.com/lifebit-ai/<replace-with-repo-name>/blob/dev/containers) folder has been updated with the necessary containers to run this pipeline

File: nextflow.config [hunk 1]
```diff
@@ -17,6 +17,7 @@ params {
   filename_hpo = ''
   sample_name = null
   config = 'conf/standard.config'
+  ped_parser_py = '/Users/ryancardenas/Documents/fix_pipelines/exomiser-pipeline-nf/bin/ped_module.py'
```
Comment (enadezhdin) [inline nextflow.config:20]: This parameter definition might not work well on external environments

File: nextflow.config [hunk 2]
```diff
@@ -61,7 +62,7 @@ process {
   errorStrategy = params.errorStrategy
   withName: exomiser {
     container = "quay.io/lifebitai/exomiser:${params.exomiser_container_tag}"
-    containerOptions = "--volume ${params.exomiser_data_directory}:/data/exomiser-data-bundle/"
+//    containerOptions = "--volume ${params.exomiser_data_directory}:/data/exomiser-data-bundle/"
```
Comment (enadezhdin) [inline nextflow.config:65]: It's better to avoid to keep commented code in merging branches

File: main.nf [hunk 3]
```diff
@@ -78,109 +84,108 @@ println(selected_prioritisers)
...
-Channel.fromPath("${params.hpo_terms_file}")
-      .splitCsv(sep: ',', skip: 1)
-      .unique()
-      .map {it -> it.toString().replaceAll("\\[", "").replaceAll("\\]", "")}
-      .map {it -> "'"+it.trim()+"'"}
-      .reduce { a, b -> "$a,$b" }
-      .into { ch_hpo_terms_file ; ch_hpo_terms_file_inspect; ch_hpo_terms }
-ch_hpo_terms_file_inspect.dump(tag:'ch_hpo_terms (retrieve_hpo_terms: false)')
+//remove
+//ch_vcf_inspect.dump(tag:'ch_vcf')
```
Comment (enadezhdin) [inline main.nf:95]: better to remove

File: main.nf [hunk 4]
```diff
@@ -78,109 +84,108 @@ println(selected_prioritisers)
...
-  tag "${vcf}-${prioritiser}"
-  publishDir "${params.outdir}/${sample_name}", mode: 'copy'
-
+  tag "${vcf_path1}"
+  publishDir "${params.outdir}/${proband_id1}", mode: 'copy'
+  maxForks 2
+  submitRateLimit = '1 / 5 m'
   input:
-  file(vcf) from ch_vcf
+  //set run_id, proband_id1, hpo, file(vcf_path1), file(vcf_index_path1), proband_sex, mother_id, father_id from ch_input
```
Comment (enadezhdin) [inline main.nf:131]: also better to remove than keep commented

**Review by enadezhdin — CHANGES_REQUESTED** [review]
Hi Ryan, as I previously mentioned, there are no big issues with the PR with exception of CI test. I like how the new added code organised and commented. This is really good. I left couple of comments, but mostly minor issues.
Another question I have - in PR description:

## Internal CloudOS Tests
Not necessary if run in Internal CloudOS CI

JOB ID: 6398d4005f2b5e015b70c523 Workspace: RR603

You provided CloudOS job like this: JOB ID: 6398d4005f2b5e015b70c523 Workspace: RR603, not a link to the job. Actually, I can not get to the job page with only this information. So, if possible, please provide a cloudOS link to the test job in the corresponding field.
Also, I looked at the `cloudos_ci_test.yml` file. This is outdated version of the config, so, expectedly it won't run properly. Assuming that proper configuration of `cloudos_ci` test is a task for a separate PR, I would propose to delete this file for now. However, you have `ci_test.yml` and it should work.
The current version of the cloudos_ci test template can be found here:
https://github.com/lifebit-ai/metagwas-nf/blob/dev/.github/workflows/internal_lifebit_cloudos_ci.yml
You should contact Erdal for more information on this.

File: nextflow.config (hunk 1 above)
Comment (Vlad-Dembrovskyi) [reply nextflow.config:20]: @sk-sahu

**Comment by sk-sahu** [comment]
@Vlad-Dembrovskyi closing this PR, as this is already accommodated in this PR and merged - https://github.com/lifebit-ai/exomiser-pipeline-nf/pull/8

**Pull request closed without merging**